pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -304,6 +304,55 @@ type SecuritySpec struct {
 	KeyRotation KeyRotationSpec `json:"keyRotation,omitempty"`
 }
 
+// ClusterSecuritySpec is the CephCluster security spec to include various security items such as kms
+type ClusterSecuritySpec struct {
+	// KeyManagementService is the main Key Management option
+	// +optional
+	// +nullable
+	KeyManagementService KeyManagementServiceSpec `json:"kms,omitempty"`
+	// KeyRotation defines options for rotation of OSD disk encryption keys.
+	// +optional
+	// +nullable
+	KeyRotation KeyRotationSpec `json:"keyRotation,omitempty"`
+
+	// CephX configures CephX key settings. More: https://docs.ceph.com/en/latest/dev/cephx/
+	// +optional
+	CephX ClusterCephxConfig `json:"cephx,omitempty"`
```

Comment:
Should we consider making this a pointer, right? Though seems like all the default values will be fine, such as generation 0 and blank rotate policy, and no need to differentiate from nil.